src/hooks/core/chat/useFetchPostNotificationHook.ts

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import AnonymousMessageRepo from '../../../service/repo/anonymousMessageRepo';
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import SignedMessageRepo from '../../../service/repo/signedMessageRepo';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+
+const useFetchPostNotificationHook = () => {
+  const {localDb, refresh} = useLocalDatabaseHook();
+
+  const saveNotifications = async (notifications: any[], fromNotificationAPI: any) => {
+    if (!localDb) return;
+    const allPromises: Promise<void>[] = notifications.map((notification) => {
+      const channelList = fromNotificationAPI(notification);
+      return channelList.saveIfLatest(localDb);
+    });
+
+    try {
+      await Promise.all(allPromises);
+      refresh('channelList');
+    } catch (e) {
+      console.log('error on saving notifications:', e);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `saveNotifications` is declared as async but it doesn't handle the case when the promises inside the `allPromises` array fail. It's a good practice to add error handling for each promise in the array to ensure that one failed promise doesn't stop the execution of others.

```diff
    const allPromises: Promise<void>[] = notifications.map((notification) => {
      const channelList = fromNotificationAPI(notification);
-     return channelList.saveIfLatest(localDb);
+     return channelList.saveIfLatest(localDb).catch(e => console.log('Error saving notification:', e));
    });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->